logging/config.go

```diff
@@ -14,12 +14,12 @@ type Options map[string]zapcore.Level
 // Config defines Logger configuration.
 type Config struct {
 	// zapcore.Level at 0 is for info level.
-	Level  zapcore.Level `yaml:"level" default:"0"`
-	Output string        `yaml:"output"`
+	Level  zapcore.Level `yaml:"level" env:"LEVEL" default:"0"`
+	Output string        `yaml:"output" env:"OUTPUT"`
 	// Interval for periodic logging.
-	Interval time.Duration `yaml:"interval" default:"20s"`
+	Interval time.Duration `yaml:"interval" env:"INTERVAL" default:"20s"`
 
-	Options `yaml:"options"`
+	Options `yaml:"options" env:"OPTIONS"`
```

Comment:
This doesn't seem to work. Passing `{"OPTIONS": "foo:debug,bar:info,buz:panic"}` results in:
```
env: parse error on field "Options" of type "logging.Options": strconv.ParseInt: parsing "debug": invalid syntax
```

However, using `{"OPTIONS": "foo:-1,bar:0,buz:4"}` sets the expected levels.

Reply:
I see three options here:

* Implement TextUnmarshaler in Options
* Submit a patch to support map-of-TextUnmarshaler
* Do nothing

Reply:
I have just toyed with implementing `TextUnmarshaler` in my [FromEnv-tests branch](https://github.com/Icinga/icinga-go-library/tree/FromEnv-tests) in <https://github.com/Icinga/icinga-go-library/commit/e9ba9f5de43b40844316b37c63f00465d8fbc9ef>. Please take a look and feel free to include this commit - or a version of it - as well.